UICHKIN-398 - Hide 'new fee/fine' menu item for dcb user

Terala-Priyanka opened the pull request:
## Purpose
[DCB] Check-in: Hide Fee/Fine action menu options (Lending library)

## Approach
<!--
 How does this change fulfill the purpose? It's best to talk
 high-level strategy and avoid code-splaining the commit history.

 The goal is not only to explain what you did, but help other
 developers *work* with your solution in the future.
-->

## Refs
[UICHKIN-398](https://issues.folio.org/browse/UICHKIN-398)
<!--
  If you have a relevant JIRA issue, add a link directly to the issue URL here.
  Example: https://issues.folio.org/browse/UICHKIN-233
-->

## Screenshots
<!-- OPTIONAL
 One picture is literally worth a thousand words. When the feature is
 an interaction, an animated GIF is best. Most of the time it helps to
 include "before" and "after" screenshots to quickly demonstrate the
 value of the feature.
-->

<!-- OPTIONAL
#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.
-->

<!-- OPTIONAL
## Learning
  Help out not only your reviewer, but also your fellow developer!
  Sometimes there are key pieces of information that you used to come up
  with your solution. Don't let all that hard work go to waste! A
  pull request is a *perfect opportunity to share the learning that
  you did. Add links to blog posts, patterns, libraries or addons used
  to solve this problem.
-->

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

Dmitriy-Litvinenko commented on src/consts.js:
```diff
@@ -56,3 +56,7 @@ export const cancelFeeClaimReturned = {
 
 export const MAX_RECORDS = '1000';
+
+export const DCB_USER = {
```
Could you please clarify this user will be created to BE and always have the same ID?

Terala-Priyanka replied:
Basically, these are the virtual patrons that are created by BE scripts only.
There can be many users with lastName "DcbSystem" and type "dcb".
Here we need check for user type as it is not obtained in the checkin response.

Dmitriy-Litvinenko commented on src/util.test.js:
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
+    };
+    expect(isDcbUser(user)).toBeTruthy();
+  });
+  it('should return false when user does not have lastName as "DcbSystem"', () => {
```
Could you please add empty line before `it`?

Terala-Priyanka replied:
done

Dmitriy-Litvinenko commented on src/util.test.js:
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
+    const user = {
+      lastName: DCB_USER.lastName
+    };
+    expect(isDcbUser(user)).toBeTruthy();
```
Could you please add empty line before `expect`?

Terala-Priyanka replied:
done

Dmitriy-Litvinenko commented on src/util.test.js:
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
+    const user = {
+      lastName: 'test'
+    };
+    expect(isDcbUser(user)).toBeFalsy();
```
Could you please add empty line before `expect`?

Terala-Priyanka replied:
done

Dmitriy-Litvinenko commented on src/util.test.js:
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
+  });
+  it('should return false when user does not have lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: 'test'
```
Could you please add comma?

Terala-Priyanka replied:
done

Dmitriy-Litvinenko commented on src/util.test.js:
```diff
@@ -318,3 +320,18 @@ describe('shouldConfirmStatusModalBeShown', () => {
+describe('isDcbUser', () => {
+  it('should return true when user has lastName as "DcbSystem"', () => {
+    const user = {
+      lastName: DCB_USER.lastName
```
Could you please add comma?

Terala-Priyanka replied:
done

Dmitriy-Litvinenko commented on src/util.js:
```diff
@@ -4,7 +4,7 @@ import {
 } from 'lodash';
 
-import { statuses } from './consts';
+import { DCB_USER, statuses } from './consts';
```
Could you please split in several line?

Terala-Priyanka replied:
done

Terala-Priyanka commented:
@Dmitriy-Litvinenko 
Fixed all your comments. Also please note that I have added changes to couple more files - FeeFineDetailsButton.js and FeeFineDetailsButton.test.js
Please re-review

Dmitriy-Litvinenko commented on src/util.js:
```diff
@@ -4,7 +4,10 @@ import {
-import { statuses } from './consts';
+import {
+  DCB_USER,
+  statuses
```
Could you please add comma?

Terala-Priyanka replied:
done

Dmitriy-Litvinenko commented on src/components/FeeFineDetailsButton/FeeFineDetailsButton.js:
```diff
@@ -15,6 +15,7 @@ const FEE_FINES_STATUSES = {
 
 class FeeFineDetailsButton extends React.Component {
   static propTypes = {
+    isVirtualUser: PropTypes.bool,
```
Could you please clarify should we add `isRequired `or add default value?

Terala-Priyanka replied:
added default prop value
